### PR TITLE
Added various beyond SK programs

### PR DESCRIPTION
**Pull request by SimonRhook**
# Description
Added various programs to compare SplashKit vs No SplashKit. These programs demonstrate the abstraction that takes place within the SplashKit SDK to hide away complexity. The No SplashKit versions essentially show what is happening in the back end of SplashKit when various functions are called. 
## Type of change

_Please delete options that are not relevant._

- [x] Addition of cpp code files

## How Has This Been Tested?

Programs have all been compiled and run to ensure they work.

## Testing Checklist

- [x] Tested in latest Chrome
- [x] Tested in latest Firefox
- [x] npm run build
- [x] npm run preview

## Checklist

_Please delete options that are not relevant._

### If involving code

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings

## Folders and Files Added/Modified

_Please list the folders and files added/modified with this pull request and delete options that are not relevant._

- Added:
  - [x] beyond-splashkit/Audio
  - [x] beyond-splashkit/InputValidation

## Additional Notes

Changes displayed are very basic. The plan is to expand this into a proper guide in the future

### CODE REVIEW

File: src/content/docs/beyond-splashkit/InputValidation/int_input_validation.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+int non_sk_convert_int_validation()
+{
+    std::cout << "This function converts user input into an integer with input validation using non Splashkit functions." << std::endl;
+
+    string x;
+    std::cout << "Please enter an integer: " << std::endl;
+
+    // Holds program in a loop until user enters a valid integer
+    while (! nosk_is_integer(x))
```
Comment (matt439) [inline src/content/docs/beyond-splashkit/InputValidation/int_input_validation.cpp:26]: Space after exclamation point is inconsistent with 3 lines below.

**Review by matt439 — CHANGES_REQUESTED** [review]
I think this guide is a great idea! My personal experience after SIT102 was that I wanted to learn more about how I could make my own games without SplashKit. However, all the source code is abstracted away, which felt like a brick wall. I would have found this guide extremely useful as I tried to understand what it takes to draw graphics to the screen.

In terms of review, I only found some minor issues regarding formatting and value types. I have no doubt that this guide will help out many students who are looking to learn lower level coding.

File: src/content/docs/beyond-splashkit/Audio/Audio.mdx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    SDL_DestroyWindow(window);
+    Mix_FreeMusic(music);
+    Mix_FreeChunk(jump);
+    Mix_FreeChunk(explosion);
+}
+
+
+void music_and_effects_sk()
+{
+    float volume = 0.5;
```
[the remark this answers is not in the document]
Comment (SimonRhook) [reply src/content/docs/beyond-splashkit/Audio/Audio.mdx:101]: This is really interesting, for some reason my splashkit install has this function as float. 
image
I originally declared volume to be double and got a typecast error which lead me to changing the variable to float.

[the remark this answers is not in the document]
Comment (matt439) [reply src/content/docs/beyond-splashkit/Audio/Audio.mdx:101]: Hmm that is strange. I copied your code into the up-to-date splashkit-core repo:
image

In `sound.h` I've got it as double:
image

[the remark this answers is not in the document]
Comment (SimonRhook) [reply src/content/docs/beyond-splashkit/Audio/Audio.mdx:101]: I did discuss this with Olivia and its a known issue. I will push the change as double is supposed to be correct

**Review by matt439 — APPROVED** [review]
All issues are now resolved. Looks great!

**Review by rmachu — APPROVED** [review]
After the change, I approve this PR. Everything looks good.

**Review by omckeon — APPROVED** [review]
I have made some changes to these guides to provide some set up instructions and updated the code into tabs and individual code files (to prepare for SK Online).

These programs were really well done, and the eye dropper program is really useful to see it all come together!

**Pull request merged**